pkg/tfbridge/auto_aliasing.go

```diff
@@ -123,6 +125,9 @@ func (info *ProviderInfo) ApplyAutoAliases() error {
 		return err
 	}
 
+	buildInfo, _ := debug.ReadBuildInfo()
+	isTfgen := buildInfo != nil && strings.Contains(buildInfo.Path, "pulumi-tfgen")
```

Comment:
I'd really prefer that our generate entry point sets some internal global variable then us trying to parse the build path.

Reply:
Yes, I'm exploring that in https://github.com/pulumi/pulumi-terraform-bridge/pull/1415, but so far I can't figure out how to only set the flag for tfgen without modifying the provider itself (which we could obviously do for AWS, but that doesn't help any of our other providers).

Reply:
I see the problem. We can leave that until the next time we change how our main function works.